### DISCUSSION TITLE
Ability to update user filters from a layout extension

### DISCUSSION THREAD

**Discussion by baptadn** [directus/directus#20645] [Feature Requests]
### Summary

It would be great to be able to develop a layout that allows for the modification of [user filters](https://github.com/directus/directus/blob/7492223d6a4655d9b02474816a6acac31eba2930/packages/extensions/src/shared/types/layouts.ts#L27). The search and filter builder is handy but may be too complex for some users to utilize effectively.

### Basic Example

The idea would be to expose [an event](https://github.com/directus/directus/blob/7492223d6a4655d9b02474816a6acac31eba2930/packages/extensions/src/shared/types/layouts.ts#L39C2-L39C6) `update:filterUser` like `update:selection`:

```tsx
interface LayoutContext {
	emit: (event: 'update:selection' | 'update:layoutOptions' | 'update:layoutQuery', ...args: any[]) => void;
}
```

### Motivation

That would allow for the easy modification of current filters through an input field or a button for quick filtering, like that :

<img width="643" alt="image" src="https://github.com/directus/directus/assets/1102595/2df4c43d-dd3f-4f3e-ae22-44522c607bec">

### Detailed Design

Use case : extend the tabular layout to add a quick filter above:

```tsx
export default defineLayout({
  id: "tabular-with-search",
  name: "Tabular with search",
  icon: "reorder",
  component: LayoutComponent,
  slots: {
    options: () => undefined,
    sidebar: () => undefined,
    actions: TabularActions,
  },
  setup(props, ctx) {
    const name = ref("tabular-search");
    const { layouts } = useExtensions();
    const tabular = layouts.value.find((layout) => layout.id === "tabular")!;

    // ✅ Allow ctx.emit("update:filterUser")

    const setup = tabular?.setup(props, ctx);

    return { ...setup, name, component: tabular.component };
  },
});
```

### Requirements List

Extend the API surface

### Drawbacks

The user can set incorrect filters and break the UI

### Alternatives

none

### Adoption Strategy

no BC

**Comment by kamatil-dev** [comment]
any update on how to implement this?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
